### PR TITLE
Added ability to specify test suffix and test prefix during project registration

### PR DESCRIPTION
**Pull request by evgeniysharapov**
Added ability to add test files prefix and suffix during project type registration. Should improve toggling between implementation and test files functionality.
-----------------

Before submitting a PR make sure the following things have been done (and denote this
by checking the relevant checkboxes):

- [x] The commits are consistent with our [contribution guidelines](../CONTRIBUTING.md)
- [x] You've added tests (if possible) to cover your change(s)
- [x] All tests are passing (`make test`)
- [x] The new code is not generating bytecode or `M-x checkdoc` warnings
- [x] You've updated the changelog (if adding/changing user-visible functionality)
- [ ] You've updated the readme (if adding/changing user-visible functionality)

Thanks!

### CODE REVIEW

File: doc/configuration.md [hunk 1]
```diff
@@ -229,6 +229,22 @@ should install and add to the PATH
 [Exuberant Ctags](http://ctags.sourceforge.net/) instead of a plain ctags, which
 ships with Emacs distribution.
 
+### Adding Custom Project Types
+If a project you are working on is recognized incorrectly or you want to add your own type of projects you can add following to your Emacs initialization code
```
Comment (bbatsov) [inline doc/configuration.md]: Add a blank line after the heading.

File: doc/configuration.md [hunk 2]
```diff
@@ -229,6 +229,22 @@ should install and add to the PATH
...
+What this does is:
+1. add your own type of project, in this case `npm` package.
+2. add a file in a root of the project that helps to identify the type, in this case it is `package.json`.
+3. add *compile-command*, in this case it is `npm build`.
+4. add *test-command*, in this case it is `npm test`.
+5. add *run-command*, in this case it is `npm start`.
+6. add test files suffix for toggling between implementation/test files, in this case it is `.spec`, so the implementation/test file pair could be `service.js`/`service.spec.js` for example.
+7. add test files prefix for toggling between implementation/test files, in this case it is `nil`. 
+
+
```
Comment (bbatsov) [inline doc/configuration.md]: This this extra blank line here.

File: projectile.el [hunk 3]
```diff
@@ -2061,16 +2061,22 @@ With a prefix ARG invalidates the cache first."
   "A hash table holding all project types that are known to Projectile.")
 
 (defun projectile-register-project-type
-    (project-type marker-files &optional compile-cmd test-cmd run-cmd)
+    (project-type marker-files &optional compile-cmd test-cmd run-cmd spec-suffix spec-prefix)
```
Comment (bbatsov) [inline projectile.el]: I'd use the term `test` instead of spec.

**Comment by bbatsov** [comment]
As the registration function becomes pretty complex with some many positional params I'd suggest switching it `cl-defun` so you can use `keyword` arguments with it.

**Comment by bbatsov** [comment]
Also - your changes have to rebased on top of the current `master` and ideally squashed together. You should also add a changelog entry for them.
